### PR TITLE
Adding global options to settings.yaml

### PR DESCRIPTION
**Pull request by adammhaile**
@sosiouxme @tbielawa 

Can now add these to settings.yaml
```
global_opts:
  rhpkg_push_timeout:
  distgit_threads:
  rhpkg_clone_timeout:

```

Has these defaults:
```
GLOBAL_OPT_DEFAULTS = {
    'distgit_threads': 20,
    'rhpkg_clone_timeout': 900,
    'rhpkg_push_timeout': 1200
}
```
Anything in settings.yaml will override.

### CODE REVIEW

**Comment by tbielawa** [comment]
Now that the settings file is becoming more involved I feel that this PR requires a documentation refresh.

Also it is a good time to start _thinking_ about adding some schema validation to that file. Then the format is officially codified and defined and verifiable.

**Comment by adammhaile** [comment]
@tbielawa ack, updated doc. 
Will figure out validation in a different PR.

File: Usage.md [hunk 1]
```diff
@@ -19,10 +19,23 @@ data_path: https://github.com/openshift/ocp-build-data.git
 
 #Sub-group directory or branch to pull build data
 group: openshift-4.0
+
+#Username for running rhpkg / brew / tito
```
Comment (tbielawa) [inline Usage.md]: Does this default to `$USER` or is it required?

**Comment by adammhaile** [comment]
It defaults to $USER

File: doozerlib/distgit.py [hunk 2]
```diff
@@ -131,7 +131,7 @@ def clone(self, distgits_root_dir, distgit_branch):
                                      'local build will be sourced from the current dist-git '
                                      'contents and not the typical GitHub source. '
                                      )
-                    cmd_list = ["timeout", "900", "rhpkg"]
+                    cmd_list = ["timeout", str(self.runtime.global_opts['rhpkg_clone_timeout']), "rhpkg"]
```
Comment (sosiouxme) [inline doozerlib/distgit.py]: can this be de-inlined like the one below?
```suggestion
                     timeout = str(self.runtime.global_opts['rhpkg_clone_timeout'])
                     cmd_list = ["timeout", timeout, "rhpkg"]
```

**Pull request merged**